pkg/compressor/gzip.go

```diff
@@ -30,7 +31,7 @@ func Decompress(data []byte) ([]byte, error) {
 	r := flate.NewReader(bytes.NewReader(data))
 	defer func() {
 		if err := r.Close(); err != nil {
-			fmt.Println("failed to close the reader")
+			log.Println("failed to close the reader")
```

Comment:
Do not use the standard `log` package, use your initialized custom logger

Reply:
It wasn't there. I thought it's not mandatory as long as the tests were passing. I added a retry mechanism for the DB within the InitPostgresDB() method. It attempts to retry 3 more times using an exponential backoff of 1s, 3s, 5s also I added a dynamic context to ensure the attempt doesn't take longer than the exponential backoff.

Reply:
I also added a try mechanism on the client so it attempts to resent the metrics using an exponential backoff of 1s, 3s, 5s if no luck then it gracefully exits the program.